### ISSUE TITLE
Stack buffer overflow in PE parser

### ISSUE THREAD

**Issue by t00sh** [radare/radare2#13062]
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Debian 4.9.130-2, AMD 64
| File format of the file you reverse (mandatory)      | PE
| Architecture/bits of the file (mandatory)            | x86-32
| r2 -v full output, **not truncated** (mandatory)     | radare2 3.3.0-git 20802 @ linux-x86-64 git.3.2.1-296-ge1f23fab4 commit: e1f23fab43c665b29029cd65af208c1d5eff13b5 build: 2019-02-10__20:24:26

### Actual behavior

There is a stack based overflow in the function `_parse_resource_directory` in the PE parser.

### Steps to reproduce the behavior 
- Compile with ASAN
- Run the attached binary to radare2 : [pe-stack-overflow.zip](https://github.com/radare/radare2/files/2855958/pe-stack-overflow.zip)

### Additional Logs, screenshots, source-code,  configuration dump, ...

The ASAN trace : 

```
==2653==ERROR: AddressSanitizer: stack-buffer-overflow on address 0x7ffe5e6bb2cf at pc 0x7fb1c3fe009e bp 0x7ffe5e6bafe0 sp 0x7ffe5e6ba
fd8                                                                                                                                   
WRITE of size 1 at 0x7ffe5e6bb2cf thread T0                      
    #0 0x7fb1c3fe009d in sdb_itoa /XXX/radare2-src-asan/shlr/sdb/src/util.c:129
    #1 0x7fb1c916da6d in _parse_resource_directory /XXX/radare2-src-asan/libr/..//$
ibr/bin/p/../format/pe/pe.c:2379                                                 
    #2 0x7fb1c916890a in _parse_resource_directory /XXX/radare2-src-asan/libr/..//$
ibr/bin/p/../format/pe/pe.c:2298                                 
    #3 0x7fb1c9171b22 in Pe64_bin_pe_parse_resource /XXX/radare2-src-asan/libr/../$
libr/bin/p/../format/pe/pe.c:2455 
```
The problem is in the `_parse_resource_directory` function, when we have a resource entry outside a directory entry : 

```C
    /* ... */
    if (resource_name) {
        rs->name = strdup (resource_name);
    } else {
        char numberbuf[6];
        rs->name = strdup (sdb_itoa (id, numberbuf, 10));
    }
    /* ... */
```
Here, the numberbuf should contains at least 64 or 65 characters (it's not very clear). The overflow happen in this line in `sdb_itoa` function : 

```C
SDB_API char *sdb_itoa(ut64 n, char *s, int base) {
    static const char* lookup = "0123456789abcdef";
     char tmpbuf[64], *os = NULL;
     const int imax = 62;

      /* ... */
      s[imax + 1] = '\0';
      /* ... */
}
```

**Comment by t00sh** [comment]
I proposed a fix in #13063.

**Comment by radare** [comment]
thanks <3
